packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -684,27 +717,48 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     private readonly chunkMap: Map<string, string[]>;
 
     private readonly dataStores: DataStores;
-    private readonly runtimeOptions: Readonly<IContainerRuntimeOptions>;
+
+    // This is the source of truth for writing in the summary metadata blob.
+    private readonly summaryEnableGC: boolean | undefined;
+    // This is the source of truth for whether GC is enabled or not.
+    private readonly shouldRunGC: boolean;
+    /**
+     * True if generating summaries with isolated channels is
+     * explicitly disabled. This only affects how summaries are written,
+     * and is the single source of truth for this container.
+     */
+    public readonly disableIsolatedChannels: boolean;
 
     private constructor(
         private readonly context: IContainerContext,
         private readonly registry: IFluidDataStoreRegistry,
         metadata: IContainerRuntimeMetadata | undefined,
         chunks: [string, string[]][],
-        runtimeOptions: IContainerRuntimeOptions = {
-            generateSummaries: true,
-        },
+        private readonly runtimeOptions: Readonly<Required<IContainerRuntimeOptions>>,
         private readonly containerScope: IFluidObject,
         public readonly logger: ITelemetryLogger,
         private readonly requestHandler?: (request: IRequest, runtime: IContainerRuntime) => Promise<IResponse>,
         private _storage?: IDocumentStorageService,
     ) {
         super();
 
-        this.runtimeOptions = {
-            ...{ disableIsolatedChannels: true },
-            ...runtimeOptions,
-        };
+        // enableGC in metadata is introduced with v1 in the metadata blob. Force to false before that.
+        // This will override the value in runtimeOptions if it is set (true/false). So setting it in
+        // runtimeOptions will only specify what to do if it has never been set before.
+        // Note that even leaving it undefined will force it to false if no metadata blob is written.
+        const prevSummaryEnableGC = context.baseSnapshot ? gcEnabled(metadata) : undefined;
```

Comment:
Is the check for context.baseSnapshot used to determine if this is a new or existing document? Can we simply say if (!context.existing) read this value from the runtime options, else get this value from the summary metdata?

Reply:
What's the benefit of using `context.existing`? I'm thinking about it purely from loaded-from summary perspective.

Reply:
I think they mean the same thing now. With detached container, existing documents that are loaded will always have a baseSnapshot.
The idea behind adding this functionality is - If you are creating a new Container, you can specify whether you want GC or not. After that, we will respect this choice whenever this container is loaded again and it cannot be changed.

So, we essentially want a check that says if this is a new container, try to get this information from runtime options. Else try to get it from the summary. I believe we use the `existing` flag to do this kind of distinction in our code.

Reply:
Okay I went ahead and changed it.